storage/engine/mvcc_test.go

```diff
@@ -1236,6 +1236,43 @@ func TestMVCCResolveTxn(t *testing.T) {
 	}
 }
 
+// TestMVCCConditionalPutOldTimestamp tests a case where a conditional
+// put with an older timestamp happens after a put with a newer timestamp.
+func TestMVCCConditionalPutOldTimestamp(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	engine := createTestEngine()
+	defer engine.Close()
+
+	err := MVCCPut(engine, nil, testKey1, makeTS(1, 0), value1, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = MVCCPut(engine, nil, testKey1, makeTS(3, 0), value2, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// QUESTION(kkaneda): Should this fail with WriteTooOldError?
```

Comment:
Thanks for taking a look, @bdarnell ! Ah, the explanation with Get/Put transaction makes sense. 

Is it OK to simply stop using `proto.MaxTimestamp` in `MVCCConditionalPut`? I don't fully understand the comment, but it seems we anyway get a `WriteTooOld` error if there is a write intent with a newer timestamp.